process/block/interceptedBlocks/interceptedBlockHeader.go

```diff
@@ -34,11 +38,12 @@ func NewInterceptedHeader(arg *ArgInterceptedBlockHeader) (*InterceptedHeader, e
 	}
 
 	inHdr := &InterceptedHeader{
-		hdr:              hdr,
-		hasher:           arg.Hasher,
-		sigVerifier:      arg.HeaderSigVerifier,
-		shardCoordinator: arg.ShardCoordinator,
-		chainID:          arg.ChainID,
+		hdr:               hdr,
+		hasher:            arg.Hasher,
+		sigVerifier:       arg.HeaderSigVerifier,
+		shardCoordinator:  arg.ShardCoordinator,
+		chainID:           arg.ChainID,
+		epochStartTrigger: arg.EpochStartTrigger,
```

Comment:
Instead  of adding in InterceptedHeader component EpochStartTrigger to can verify header epoch maybe you can modify HeaderSigVerifier component in something like HeaderVerifier and that component also checks header epoch because maybe you should check header epoch also on consensus topic.

Reply:
I thought first to apply to header validator. But here it is a special case for the epoch, which is calculated at interceptedHeader - integrity and validity according to the view of the node

Reply:
yes, we could add a "headerVerifier" that contains the HeaderSigVerifier and the EpochStartTrigger verifier, but it could be done in a refactoring in the future.